registry/client/errors.go

```diff
@@ -1,139 +1,27 @@
 package client
 
 import (
-	"encoding/json"
-	"errors"
-	"fmt"
-	"io"
-	"io/ioutil"
-	"net/http"
-
-	"github.com/distribution/distribution/v3/registry/api/errcode"
-	"github.com/distribution/distribution/v3/registry/client/auth/challenge"
+	"github.com/distribution/distribution/v3/registry/client/errors"
 )
 
 // ErrNoErrorsInBody is returned when an HTTP response body parses to an empty
 // errcode.Errors slice.
-var ErrNoErrorsInBody = errors.New("no error details found in HTTP response body")
+var ErrNoErrorsInBody = errors.ErrNoErrorsInBody
```

Comment:
I'd  suggest

- deprecating the old ones (with a `Deprecated` comment pointing to the new location)
-  renaming the file from `errors.go` to `errors_deprecated.go` (e.g.) to emphasise that no new errors should be added.
- given that they'd be deprecated, change existing uses in the code to use the new package (`HandleErrorResponse` and `SuccessStatus` look to be used in various locations)
- Now that the file only contains aliases, perhaps group the vars and types in a `var(..)` and `type(...)` block.

Quick write-up for that;

```go
package client

import "github.com/distribution/distribution/v3/registry/client/errors"

var (
	// ErrNoErrorsInBody is returned when an HTTP response body parses to an empty
	// errcode.Errors slice.
	//
	// Deprecated: use github.com/distribution/distribution/v3/registry/client/errors.ErrNoErrorsInBody
	ErrNoErrorsInBody = errors.ErrNoErrorsInBody

	// HandleErrorResponse returns error parsed from HTTP response for an
	// unsuccessful HTTP response code (in the range 400 - 499 inclusive). An
	// UnexpectedHTTPStatusError returned for response code outside of expected
	// range.
	//
	// Deprecated: use github.com/distribution/distribution/v3/registry/client/errors.HandleErrorResponse
	HandleErrorResponse = errors.HandleErrorResponse

	// SuccessStatus returns true if the argument is a successful HTTP response
	// code (in the range 200 - 399 inclusive).
	//
	// Deprecated: use github.com/distribution/distribution/v3/registry/client/errors.SuccessStatus
	SuccessStatus = errors.SuccessStatus
)

type (
	// UnexpectedHTTPStatusError is returned when an unexpected HTTP status is
	// returned when making a registry api call.
	//
	// Deprecated: use github.com/distribution/distribution/v3/registry/client/errors.UnexpectedHTTPStatusError
	UnexpectedHTTPStatusError = errors.UnexpectedHTTPStatusError

	// UnexpectedHTTPResponseError is returned when an expected HTTP status code
	// is returned, but the content was unexpected and failed to be parsed.
	//
	// Deprecated: use github.com/distribution/distribution/v3/registry/client/errors.UnexpectedHTTPResponseError
	UnexpectedHTTPResponseError = errors.UnexpectedHTTPResponseError
)
```